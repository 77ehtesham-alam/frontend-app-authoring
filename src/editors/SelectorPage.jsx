import React from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import ErrorBoundary from './sharedComponents/ErrorBoundary';
import { Selector } from './Selector';
import store from './data/store';

const SelectorPage = ({
  courseId,
  lmsEndpointUrl,
  studioEndpointUrl,
}) => (
  <ErrorBoundary>
    <Provider store={store}>
      <Selector
        {...{
          learningContextId: courseId,
          lmsEndpointUrl,
          studioEndpointUrl,
        }}
      />
    </Provider>
  </ErrorBoundary>
);

SelectorPage.defaultProps = {
  courseId: null,
  lmsEndpointUrl: null,
  studioEndpointUrl: null,
};

SelectorPage.propTypes = {
  courseId: PropTypes.string,
  lmsEndpointUrl: PropTypes.string,
  studioEndpointUrl: PropTypes.string,
};

export default SelectorPage;
