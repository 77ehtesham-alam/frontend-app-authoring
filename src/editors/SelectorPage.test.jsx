import React from 'react';
import { shallow } from 'enzyme';
import SelectorPage from './SelectorPage';

const props = {
  courseId: 'course-v1:edX+DemoX+Demo_Course',
  lmsEndpointUrl: 'evenfakerurl.com',
  studioEndpointUrl: 'fakeurl.com',
};

jest.mock('react-redux', () => ({
  Provider: 'Provider',
}));
jest.mock('./Selector', () => 'Selector');

describe('Selector Page', () => {
  describe('snapshots', () => {
    test('rendering correctly with expected Input', () => {
      expect(shallow(<SelectorPage {...props} />)).toMatchSnapshot();
    });
    test('rendering with props to null', () => {
      expect(shallow(<SelectorPage />)).toMatchSnapshot();
    });
  });
});
