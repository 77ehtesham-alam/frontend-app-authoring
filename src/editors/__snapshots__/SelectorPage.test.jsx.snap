// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector Page snapshots rendering correctly with expected Input 1`] = `
<ErrorBoundary>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(Symbol.observable): [Function],
      }
    }
  >
    <Component
      learningContextId="course-v1:edX+DemoX+Demo_Course"
      lmsEndpointUrl="evenfakerurl.com"
      studioEndpointUrl="fakeurl.com"
    />
  </Provider>
</ErrorBoundary>
`;

exports[`Selector Page snapshots rendering with props to null 1`] = `
<ErrorBoundary>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(Symbol.observable): [Function],
      }
    }
  >
    <Component
      learningContextId={null}
      lmsEndpointUrl={null}
      studioEndpointUrl={null}
    />
  </Provider>
</ErrorBoundary>
`;
